src/ArborX_HDBSCAN.hpp

```diff
@@ -12,31 +12,44 @@
 #define ARBORX_HDBSCAN_HPP
 
 #include <ArborX_Dendrogram.hpp>
+#include <ArborX_DetailsKokkosExtScopedProfileRegion.hpp>
 #include <ArborX_MinimumSpanningTree.hpp>
 
 namespace ArborX::Experimental
 {
 
 template <typename ExecutionSpace, typename Primitives>
 auto hdbscan(ExecutionSpace const &exec_space, Primitives const &primitives,
-             int core_min_size)
+             int core_min_size,
+             DendrogramImplementation dendrogram_impl =
+                 DendrogramImplementation::DEFAULT)
```

Comment:
If there is a default value, I don't see a reason to have an additional `DEFAULT` option. If users want to use the default, they would omit the argument anyway. What was your rational for introducing it?